### PR TITLE
Antrea Project Security Self-Assessment - Security Pals

### PR DESCRIPTION
**Pull request by molofgarb**
Created and added first draft for Antrea Project Security Self-Assessment.

We are still finalizing the implementation of feedback from the Antrea maintainers for a couple of sub-sections. The PR will be updated by force-pushes as we finish our revisions.

Please feel free to share your feedback on the security self-assessment!

### CODE REVIEW

**Comment by eddie-knight** [comment]
Hi there, and thanks for your work on this self assessment!

I noticed that you included an SBOM along with the self assessment. There are two reasons that jump to the front of my mind for why this isn't needed...

1. SBOMs should be associated with releases, as the bill of materials is only accurate and useful if it is created at build time and associated to a particular point in the code history.
1. If you need to link to an SBOM for some reason in the self-assessment, you can just provide a link out to the latest build artifacts that contain an SBOM.

We still have plenty more to review, but as a starter— could you please remove the SBOM from this PR?

**Comment by molofgarb** [comment]
> Hi there, and thanks for your work on this self assessment!
> 
> I noticed that you included an SBOM along with the self assessment. There are two reasons that jump to the front of my mind for why this isn't needed...
> 
>     1. SBOMs should be associated with releases, as the bill of materials is only accurate and useful if it is created at build time and associated to a particular point in the code history.
> 
>     2. If you need to link to an SBOM for some reason in the self-assessment, you can just provide a link out to the latest build artifacts that contain an SBOM.
> 
> 
> We still have plenty more to review, but as a starter— could you please remove the SBOM from this PR?

Hi Eddie, thank you for your feedback! We initially included the SBOM because a section for the SBOM is present in the [self-assessment template](https://github.com/cncf/tag-security/blob/main/assessments/guide/self-assessment.md). Your points are right though and we have removed the SBOM and its section in the self-assessment from our latest force push.

**Review by ragashreeshekar — CHANGES_REQUESTED** [review]
Thanks for the PR @molofgarb and team, appreciate the efforts.
I have completed first pass of review. Please feel free to reach out here or on slack for any questions and clarifications.

Along with addressing the comments, kindly update the PR branch with the latest content in the repo as this branch is out-of-date with the base branch.

**Comment by molofgarb** [comment]
In our previous push, we have added a table to the "Related Projects and Vendors" section to briefly compare CNI projects.

**Comment by molofgarb** [comment]
Thank you for your feedback @ragashreeshekar! We will work on adding your suggested changes and let you know once we have resolved them.

File: assessments/projects/antrea/self-assessment.md [hunk 1]
```diff
@@ -0,0 +1,506 @@
...
+* [Security issue resolution](#security-issue-resolution)
+* [Appendix](#appendix)
+
+## Metadata
+
+|   |  |
+| -- | -- |
+| Software | https://github.com/antrea-io/antrea |
+| Security Provider | No  |
+| Languages | Go, Shell, PowerShell, Makefile, Dockerfile, Jinja |
```
[the remark this answers is not in the document]
Comment (molofgarb) [reply assessments/projects/antrea/self-assessment.md:32]: We have provided a link to a website where an SBOM for the current version of Antrea can be downloaded

File: assessments/projects/antrea/self-assessment.md [hunk 2]
```diff
@@ -0,0 +1,506 @@
...
+viii. **Network Flow Visibility:**
+* Antrea supports exporting [network flow information using IPFIX](https://antrea.io/docs/v1.2.3/docs/getting-started/), which can be visualized with Elastic Stack and Kibana dashboards. This visibility enhances security monitoring capabilities.
+
+ix. **Web UI Security:**
+* The Antrea web UI, offering a graphical interface for monitoring, is subject to secure access control facilitated by Kubernetes RBAC. Permissions for the UI server, including API access, are provided through K8s RBAC, while user access to the UI is governed by either password-based or OIDC-based authentication methods. This dual-layered approach ensures both API-level security and user authentication, contributing to a secure monitoring experience.
+
+x. **Integrations with IPS/IDS:**
+* Antrea can be [integrated](https://github.com/antrea-io/antrea/tree/main/docs/cookbooks/ids) with IDS and IPS engines. The process works by configuring a TrafficControl resource applying to specific Pods. Traffic originating from the Pods or destined for the Pods is mirrored, and then inspected by an IDS/IPS to provide threat detection.
+
+### Threat Model
```
[the remark this answers is not in the document]
Comment (molofgarb) [reply assessments/projects/antrea/self-assessment.md]: The threat model has been moved to the appendix

File: assessments/projects/antrea/self-assessment.md [hunk 3]
```diff
@@ -0,0 +1,506 @@
...
+  [Kubernetes upstream tests](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-testing/e2e-tests.md).
+  * Daily and weekly Jenkins tests are automatically run on the main branch of 
+    the Antrea repository. Developers are informed of test failures through
+    the
+    [developer mailing list](https://groups.google.com/forum/#!forum/projectantrea-dev).
+* Using a 
+  [DCO GitHub App](https://github.com/apps/dco) 
+  to check that a DCO is present in every commit of a PR before review of that 
+  PR begins.
+* Running a GitHub Action to
```
[the remark this answers is not in the document]
Comment (molofgarb) [reply assessments/projects/antrea/self-assessment.md]: In that section, we mentioned that Antrea uses CodeQL, a SAST tool, and go-licenses, a SCA tool. We modified the section to explicitly mention those tools as SAST/SCA tools and organize it in a more readable manner. We also made an addition that states that Antrea does not have a secrets scanning service.

Please let us know if we have sufficient detail on the SAST/SCA/secrets scanning tools and if you think that it has been presented clearly.

File: assessments/projects/antrea/self-assessment.md [hunk 4]
```diff
@@ -0,0 +1,506 @@
...
+
+Open vSwitch (OVS): At the heart of the data plane operations lies the Open vSwitch (OVS), a high-performance, multilayer virtual switch. OVS is responsible for forwarding network packets based on the configurations and policies set by the Antrea Agents. It forms the backbone of the network’s data plane, efficiently handling network traffic within the cluster.
+
+### Actions
+In a Kubernetes cluster, Antrea initializes the Antrea Agent and the Antrea Controller. The Antrea Agent sets up Open vSwitch (OVS) bridges on each Kubernetes Node to create connections between Pods and manages the IP address allocation. The Antrea Controller watches for changes in the NetworkPolicy, Pod, and Namespace resources from the Kubernetes API. Based on instructions from the Antrea Controller, the Antrea Agent enforces network policies locally by creating OVS flows on each Node.
```
[the remark this answers is not in the document]
Comment (skyekim) [reply assessments/projects/antrea/self-assessment.md]: Thank you for the feedback. I have added more details to the Action section.

File: assessments/projects/antrea/self-assessment.md [hunk 5]
```diff
@@ -0,0 +1,506 @@
...
+
+### Goals
+The primary goal of Antrea is to enhance network connectivity and security within Kubernetes clusters. Antrea is designed to ensure efficient and secure communication between containers, utilizing strict network policies to permit only authorized traffic. It leverages Open vSwitch for high-performance networking, making it well-suited for extensive deployments. In terms of security, while Antrea focuses on network-level security, it does not automatically encrypt pod network traffic across the cluster. Instead, users have the option to enable IPsec or Wireguard encryption for the overlay network. This feature is optional and is not enabled by default. For control communications, such as from the Controller to the Agent, Antrea always uses TLS to ensure secure data transmission. However, it's important to note that Antrea's security measures are specific to network-level concerns and do not address broader security issues like application vulnerabilities. Overall, Antrea aims to provide a robust and secure networking solution for Kubernetes environments.
```
[the remark this answers is not in the document]
Comment (molofgarb) [reply assessments/projects/antrea/self-assessment.md]: We have revised this section to focus on the goals of Antrea.

**Comment by molofgarb** [comment]
Hi @torinvdb, thank you for your suggestions! We have added your changes to our most recent push.

**Review by ragashreeshekar — APPROVED** [review]

**Review by JustinCappos — APPROVED** [review]
Looks ready to merge!

**Pull request merged**